symengine/polynomial.cpp

```diff
@@ -274,8 +282,8 @@ RCP<const UnivariateIntPolynomial> mul_poly(RCP<const UnivariateIntPolynomial> a
     a1 <<= N;
     integer_class a2 = a1 / 2;
     integer_class mask = a1 - 1;
-    integer_class sa = a->eval_bit(N);
-    integer_class sb = b->eval_bit(N);
+    integer_class sa = a_sign * a.eval_bit(N);
```

Comment:
Tests don't pass when I tried changing it. I think it requires the highest degree term to turn out positive maybe?